core/src/panic.rs

```diff
@@ -8,6 +9,23 @@ pub fn configure_panic() {
         default_hook(panic_info);
         println!("Exiting...");
         // TODO: setup a wait time and fold the log system properly
```

Comment:
There is also this comment, it's probably partially addressed, but I'm not sure about the case when the logger is not initialized yet and also what's meant by the wait time.

Reply:
I think this means we should make sure that logs are actually flushed into the files before the process exits. This can possibly be done by making special calls to the underlying log library (maybe @tiram88 would know how), or at least adding a short sleep here

Reply:
Interesting, it's also the requirement I haven't clarified yet. When is it ok to lose the logs?

Even if we try to wait, the logger might never be initialized as the panic condition might happen before.
Another approach is probably to initialize a fallback logger within the hook or even write to the log file directly if needed.
But, it can cause a race condition, as the logger system can only be initialized once ([link](https://docs.rs/log/latest/log/#warning)), unless we use the different mechanism for the fallback.
But, even with this, logs still will be lost in cases like no disk space or potentially the lack of resources/permissions.

Also, the default behavior of the `log` facade is to ignore failures writing logs:

* https://github.com/rust-lang/log/issues/382
* https://github.com/rust-lang/log/issues/588

And it seems that there is not much can be done other than deciding whether to panic on log failures globally or not. (Unless we can rely on the specific loggers implementations, but more research needed.)
It's not very helpful in this case, as panicking on a log failure inside the panic itself should causes an infinite recursion code flow.

So, what would be the reasonable approach and what trade offs are acceptable?

Reply:
I'd probably propose to keep this implementation, but also write the panic error to the dedicated file.
In this case, we write to logs when the logger is available, and we write the panic file regardless of the logger status (as long as we can write to the file system).
The issue here is it probably needs some standardization, like location of the file and its content/structure, as it might also be helpful to extend it to the panic report of sorts, providing additional details about the host.
But, I'm not too sure how suitable this approach is.